src/middlewares/build-include.ts

```diff
@@ -0,0 +1,84 @@
+import { Response, NextFunction } from "express";
+import { SequelizeQueryParserRequestInterface, IncludeObject } from "../core/interfaces";
+import {
+  INVALID_INCLUDE,
+  MODEL_NOT_CONFIGURED_ERROR,
+  SEQUELIZE_QUERY_PARSER_DATA_NOT_FOUND_ERROR,
+} from "../core/constants";
+import { parseStringWithParams } from "../utils";
+import { hasIn } from "lodash";
+
+/**
+ * Middleware to build the include array for Sequelize queries based on request parameters.
+ * @param req - Express request object with SequelizeQueryParserRequestInterface
+ * @param res - Express response object
+ * @param next - Express next function
+ * @throws {Error} SEQUELIZE_QUERY_PARSER_DATA_NOT_FOUND_ERROR if necessary Sequelize query parser data is missing
+ * @throws {Error} MODEL_NOT_CONFIGURED_ERROR if the model is not configured in the request
+ */
+export function buildInclude(
+  req: SequelizeQueryParserRequestInterface,
+  res: Response,
+  next: NextFunction
+) {
+  // Check if necessary Sequelize query parser data exists
+  if (!req.sequelizeQueryParser)
+    throw new Error(SEQUELIZE_QUERY_PARSER_DATA_NOT_FOUND_ERROR);
+  if (!req.sequelizeQueryParser.model)
+    throw new Error(MODEL_NOT_CONFIGURED_ERROR);
+
+  const model = req.sequelizeQueryParser.model;
+  let include;
+  if (!req.query.include) {
+    include = [];
+  } else {
+    let valid = false;
+    const includes = (req.query.include as string).split(",");
+    includes.forEach((includeItem) => {
+      const checkInclude = includeItem.split(".").join(".target.associations.");
+      valid = hasIn(model, "associations." + checkInclude);
+      if (!valid) {
+        throw new Error(parseStringWithParams(INVALID_INCLUDE, model.name));
```

Comment:
Invalid association path 'path.in.query' for model 'modelName'

Reply:
path.in.query is includeItem in this case

Reply:
Instructions unclear. Please be more specific.

Reply:
The error message format